File: Practica Teoria/MyApplicationClase03CUM/MyApplicationClase03CUM/app/src/main/java/com/example/myapplicationclase03_cum/FrmCum.java
package com.example.myapplicationclase03_cum;

import android.content.Intent;
import android.os.Bundle;
import android.os.PersistableBundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class FrmCum extends AppCompatActivity {
    Float cum;
    Integer uv;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_cum);

        TextView txtSalidaNombre = (TextView) findViewById(R.id.TxtSalidaNombre);
        TextView txtSalidaCarnet = (TextView) findViewById(R.id.TxtSalidaCarnet);
        TextView txtSalidaCum = (TextView) findViewById(R.id.TxtSalidaUv);
        TextView txtSalidaUv = (TextView) findViewById(R.id.TxtSalidaCum);
        final Button btnCumSalida = (Button) findViewById(R.id.BtnSalidaCum);

        Bundle bundle = this.getIntent().getExtras();

        txtSalidaNombre.setText("Nombre: " + bundle.getString("NOMBRE"));
        txtSalidaCarnet.setText("Carnet: " + bundle.getString("CARNET"));
        txtSalidaCum.setText("CUM: " + bundle.getString("CUM"));

        cum = Float.parseFloat(bundle.getString("CUM"));

        if (cum >= 7.5) uv = 32;
        else if (cum >= 7.00 && cum < 7.5) uv = 24;
        else if (cum >= 6.0 && cum < 7.0) uv = 20;
        else uv = 16;

        txtSalidaUv.setText("UV: " + uv.toString());

        btnCumSalida.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(FrmCum.this, MainActivity.class);
                startActivity(intent);
            }
        });

    }

}

File: Practica-Desafio-01/MyApplicationPracticaDesafio01ReyesPineda/MyApplicationPracticaDesafio01ReyesPineda/app/src/main/java/com/example/myapplicationpracticadesafio01reyespineda/FrmCum.kt
package com.example.myapplicationpracticadesafio01reyespineda

import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class FrmCum : AppCompatActivity() {

    /**
     * Este formulario mostrara la informacion procesada del formulario 01 xD
     *
     */

    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_frm_cum)

        var txtSalidaNombre = TextView(this)
        var txtSalidaCarnet = TextView(this)
        var txtSalidaCum = TextView(this)
        var txtSalidaUv = TextView(this)
        var btnCumSalida = TextView(this)

        txtSalidaNombre = findViewById(R.id.TxtSalidaNombre)
        txtSalidaCarnet = findViewById(R.id.TxtSalidaCarnet)
        txtSalidaCum = findViewById(R.id.TxtSalidaCum)
        txtSalidaUv = findViewById(R.id.TxtSalidaUv)
        btnCumSalida = findViewById(R.id.BtnSalidaCum)

        val extras = intent.extras

        txtSalidaNombre.text = extras?.getString("nombre")
        txtSalidaCarnet.text = extras?.getString("carnet")
        txtSalidaCum.text = extras?.getString("cum")

        var cum:Float = txtSalidaCum.text.toString().toFloat()
        var uv:Float

        if (cum >= 7.5 ) uv = 32F;
        else if (cum >= 7.0 && cum < 7.5) uv = 24F;
        else if (cum >= 6.0 && cum < 7.0) uv = 20F;
        else uv = 16F;

        txtSalidaUv.text = uv.toString();

        btnCumSalida.setOnClickListener{
            val intent = Intent(this, MainActivity::class.java)
            startActivity(intent)
        }
    }

        fun rationalFromDouble(d: Double): Pair<Int, Int> {
            val s = d.toString()
            val digitsDec = s.length - 1 - s.indexOf('.')

            var tempD = d
            var tempDenom = 1
            repeat(digitsDec) {
                tempD *= 10
                tempDenom *= 10
            }
            val num = tempD.toInt()

            return Pair(num, tempDenom)
        }

   // val d = 3.14
   // val (numerator, denominator) = rationalFromDouble(d)
   // println("Numerator: $numerator, Denominator: $denominator")

}